src/serializer/ResidualHeapVisitor.js

```diff
@@ -561,6 +558,23 @@ export class ResidualHeapVisitor {
     }
   }
 
+  _visitBindingHelper(residualFunctionBinding: ResidualFunctionBinding) {
+    let environment = residualFunctionBinding.declarativeEnvironmentRecord;
+    invariant(environment !== null);
+    if (residualFunctionBinding.value === undefined) {
+      // The first time we visit, we need to initialize the value to its equivalent value
```

Comment:
Please put a comment here about why you're delaying initializing the value of the residualFunctionBinding until here. I suspect it has to do with making sure effects are applied, but it would be nice to document that somewhere.

Reply:
Not sure what you mean by "delaying". `_visitBindingHelper` will get called right away by `visitBinding` through the code you wrote.